### DISCUSSION TITLE
New to Livewire 3, inline scripts do not work, no error, doesnt trigger event

### DISCUSSION THREAD

**Discussion by narutodiablo** [livewire/livewire#7555] [1. Help] [answered]
New to Livewire 3 and needing to use inline javascripts but its not working. 

on my layout.blade.php , i have added stack('scripts')

on my livewire component partial i have added 

```
@push('scripts')
<script>
    document.addEventListener('livewire:init', function () {
     console.log('loaded');
      });
</script>
@endpush
```

I also tried other events like DOMContentLoaded but still doesnt work.

I have been here for a few hours trying to solve this basic inline javascript. can anyone please help me?

**Comment by joshhanley** [comment]
@narutodiablo hmm that should work. Can you share what your layout looks like?

**Reply by narutodiablo** [reply, marked as answer]
hello @joshhanley thanks for the response. sorry for the delay since i only saw your response.

i managed to fix this issue by updating livewire from 3.1 to 3.3. the @push and @endpush were being treated as literals in the blade in livewire 3.1 so i was stumped for a few days.

it was just fortunate that i came across some post somewhere that mentioned that updating from version 3.1 fixes this issue. thanks anyway!

**Comment by nbyloff** [comment]
I am on version 3.4.7. This inline script is in a livewire component that is inside a filament widget. Livewire component displays properly and the following  prints "started" but not "loaded" in the console. No errors. Same result if I try with the `@script` and `@endscript` tags

````
@push('scripts')
    <script>
        console.log('started');
        document.addEventListener('livewire:init', function () {
            console.log('loaded');
        });
    </script>
@endpush

````

**Reply by dostogirmir** [reply]
@nbyloff Please check this i think it help you.

https://livewire.laravel.com/docs/javascript#:~:text=To%20execute%20bespoke%20JavaScript%20in,the%20execution%20of%20this%20JavaScript.


![image](https://github.com/livewire/livewire/assets/32570245/d6190115-0957-4d1b-b22a-8936e912e8e0)

**Comment by AhmedSewidan** [comment]
``
    I found that this is the best solution to handle all scripts that faild after and livewire action or movement ✅✅
    
    document.addEventListener('livewire:init', () => {
        Livewire.hook('morphed', ({
            el,
            component
        }) => {
            setTimeout(() => {
                initializeAll();
            }, 500);
        })
    });
    
    Try and if it worked react on me🎯
``
